### DISCUSSION TITLE
False alarm conflicts & Error when running job render-page after 2.5.272 -> 2.5.287 upgrade

### DISCUSSION THREAD

**Discussion by JanZeman** [requarks/wiki#5697] [Error / Bug Report] [answered]
Hello,

After I upgraded my wiki from 2.5.272 -> 2.5.287 I started to get 2 (probably related) errors.

![image](https://user-images.githubusercontent.com/1670470/193412023-c9bdc058-b52d-4ce0-9d5e-b50229af6697.png)

![image](https://user-images.githubusercontent.com/1670470/193412152-0362df55-bc0d-4943-8078-2ee6c8a4e051.png)

The info about the edit conflict is  a false alarm. I get it for example if I try to save a page that I edited as the very same user in the very same browser few hours ago. Nobody edited it in the meantime. I save some page, do not close it, come to the same still opened browser in few hours, press save button and the conflict appears.

The second bug is even a bigger obstacle. The given page stop accepting any changes.

The only solution is to manually login to my hosting server and restart the node.js process. Then I have to 'resolve the conflict' and can continue editing the page.

Is this a known problem? Can anyone help what to do? The need of wiki restart makes it very hard for daily usage...

**EDIT:** Actually the same problems happen even when I tried to create a completely new page 😥

**Comment by JanZeman** [comment]
Hi Nicolas @NGPixel , I apologize to tag your name but last time you were the only one who reacted. On this ticket I have waited for 10 days but nobody reacted so I am trying at least this way. The problem described above makes my daily work with the wiki almost impossible. I basically cannot edit anything or add new page - everything ends up with the huge red error message above. Is there anything I can do? Should I try to upgrade? Or downgrade? Downgrading might be dangerous, I guess (DBS changes). I am bit afraid of the upgrade as that is how the problems started but if it could help...

**Reply by NGPixel** [reply]
Make sure your wiki has proper write permissions on the data folder. If you have any storage module enabled, try turning them off and try again.

**Reply by JanZeman** [reply]
Thank you for your reply @NGPixel. You mean the Linux permissions, right? It shows **drwxr-xr-x**. I am not very strong in Linux but yes, this sounds logical. I understand the first w stands for write permissions for the owner - which is the user I normally log in to the maintenance terminal - so the account I copied the updated wiki version files under. If I go into data folder I find this: drwxrwxr-x cache, drwxr-xr-x content, drwxrwxr-x uploads. So those subfolders have write permission also for the group (with exception of the content subfolder). And I remember I setup cron job to start the individual node processes which I guess runs under other than my account.

I found Chmod Recursive documentation. **Should I run anything like 'chmod -R drwxrwxr-x ./data'?** So changing permissions for the data folder and its entire content? Can I anywhere see what are expected permissions here? I am surprised I see less permissions on content subfolder compared to upload one.

I do not have any active storage target. My 2 wikis installed on the same server have minimum settings compared to freshly installed wiki.js.

**Comment by JanZeman** [comment, marked as answer]
I installed 3rd 'test' wiki instance. Observed in detail its data folder and its subfolders permissions. I checked it on my 2 existing wiki instances and did not find any difference. I decided to try overriding these permission but before doing so I upgraded to 2.5.289 version. I did it the same way I did the previous update to 2.5.287 - at least I believe so - and **after this upgrade the problem disappeared**. 

I do not think the fix is the 2.5.289 version itself. To be more accurate next time and to hopefully safe hours of my time I scripted the upgrade process. So hopefully my next update will be smooth...
